In [1]:
import pandas as pd
import numpy as np

## Loading diabetes dataset ##

In [2]:
df = pd.read_csv("diabetes.csv")

In [3]:
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
df.shape

(768, 9)

In [5]:
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [6]:
df['Outcome'].value_counts()

0    500
1    268
Name: Outcome, dtype: int64

from the above it means that 500 of them are non diabetic and 268 of them are diabetic. 

In [7]:
df.groupby('Outcome').mean()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
Outcome                                                                      
0           3.298000  109.980000      68.184000      19.664000   68.792000   
1           4.865672  141.257463      70.824627      22.164179  100.335821   

               BMI  DiabetesPedigreeFunction        Age  
Outcome                                                  
0        30.304200                  0.429734  31.190000  
1        35.142537                  0.550500  37.067164

## Now we will seperate the data into data and labels ##

In [8]:
X = df.drop(columns='Outcome',axis=1)
y = df['Outcome']

In [9]:
print(X)

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  


In [10]:
print(y)

0      1
1      0
2      1
3      0
4      1
      ..
763    0
764    0
765    0
766    1
767    0
Name: Outcome, Length: 768, dtype: int64


## Since the given above values have different standards like pregnencies, bmi, B.p , so we will standerised these values. ##

In [11]:
from sklearn.preprocessing import StandardScaler

In [12]:
scaler = StandardScaler()

In [13]:
scaler.fit(X)

StandardScaler()

In [14]:
new_df = scaler.transform(X)

In [15]:
print(new_df)

[[ 0.63994726  0.84832379  0.14964075 ...  0.20401277  0.46849198
   1.4259954 ]
 [-0.84488505 -1.12339636 -0.16054575 ... -0.68442195 -0.36506078
  -0.19067191]
 [ 1.23388019  1.94372388 -0.26394125 ... -1.10325546  0.60439732
  -0.10558415]
 ...
 [ 0.3429808   0.00330087  0.14964075 ... -0.73518964 -0.68519336
  -0.27575966]
 [-0.84488505  0.1597866  -0.47073225 ... -0.24020459 -0.37110101
   1.17073215]
 [-0.84488505 -0.8730192   0.04624525 ... -0.20212881 -0.47378505
  -0.87137393]]


In [16]:
X = new_df

In [17]:
y = df['Outcome']

In [18]:
print(X)

[[ 0.63994726  0.84832379  0.14964075 ...  0.20401277  0.46849198
   1.4259954 ]
 [-0.84488505 -1.12339636 -0.16054575 ... -0.68442195 -0.36506078
  -0.19067191]
 [ 1.23388019  1.94372388 -0.26394125 ... -1.10325546  0.60439732
  -0.10558415]
 ...
 [ 0.3429808   0.00330087  0.14964075 ... -0.73518964 -0.68519336
  -0.27575966]
 [-0.84488505  0.1597866  -0.47073225 ... -0.24020459 -0.37110101
   1.17073215]
 [-0.84488505 -0.8730192   0.04624525 ... -0.20212881 -0.47378505
  -0.87137393]]


In [19]:
print(y)

0      1
1      0
2      1
3      0
4      1
      ..
763    0
764    0
765    0
766    1
767    0
Name: Outcome, Length: 768, dtype: int64


# Now it comes over training and testing our data. ##

In [20]:
from sklearn.model_selection import train_test_split
X_train , X_test , y_train , y_test = train_test_split(X,y,test_size = 0.2, stratify = y, random_state = 2)

In [21]:
print(X.shape)

(768, 8)


In [22]:
print(X_train.shape)

(614, 8)


In [23]:
print(X_test.shape)

(154, 8)


In [24]:
## Now we will use svm support vector machine for the classificaton. ##

In [25]:
from sklearn import svm

In [26]:
model1 = svm.SVC(kernel = 'linear')

In [27]:
model1.fit(X_train, y_train)

SVC(kernel='linear')

## Model Evaluation & Accuracy score . ##

In [28]:
from sklearn.metrics import accuracy_score

In [29]:
X_predicted = model1.predict(X_train)
training_data_accuracy = accuracy_score(X_predicted, y_train)

In [30]:
print("Accuracy of the training data is : ", training_data_accuracy)

Accuracy of the training data is :  0.7866449511400652


In [31]:
##Now the accuracy prediction of test data. ##

In [32]:
X_test_predicted = model1.predict(X_test)
training_data_accuracy = accuracy_score(X_test_predicted, y_test)

In [33]:
print("Accuracy of the test data is : ", training_data_accuracy)

Accuracy of the test data is :  0.7727272727272727


In [34]:
import pickle

In [35]:
filename = 'diabetes_model.sav'
pickle.dump(model1, open(filename, 'wb'))

In [36]:
loaded_model = pickle.load(open('diabetes_model.sav', 'rb'))

## Now we will make a predictive system. ##

In [37]:
input = (4,110,92,0,0,37.6,0.191,30)

In [38]:
input_numpy = np.asarray(input)

In [39]:
input_data = input_numpy.reshape(1,-1)

In [40]:
std_data = scaler.transform(input_data)

C:\Users\Arpit\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  warnings.warn(


In [41]:
print(std_data)

[[ 0.04601433 -0.34096773  1.18359575 -1.28821221 -0.69289057  0.71168975
  -0.84827977 -0.27575966]]


In [42]:
prediction = model1.predict(std_data)

In [43]:
print(prediction)

[0]


In [44]:
if (prediction[0] == 0):
    print("Non-diabetic person")
else:
    print("Diabetic person")

Non-diabetic person
